src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -55,144 +56,43 @@
-      startReadOnlyTransaction(connection);
-
-      String stairwayId = lookupStairwayInstanceQuery(connection, stairwayName);
-
-      commitTransaction(connection);
-      return stairwayId;
-    } catch (SQLException ex) {
-      throw new DatabaseOperationException("Stairway instance lookup failed", ex);
-    }
-  }
-
-  private String lookupStairwayInstanceQuery(Connection connection, String stairwayName)
-      throws SQLException, DatabaseOperationException {
-    final String sqlStairwayInstance =
-        "SELECT stairway_id"
-            + " FROM "
-            + STAIRWAY_INSTANCE_TABLE
-            + " WHERE stairway_name = :stairwayName";
-
-    try (NamedParameterPreparedStatement instanceStatement =
-        new NamedParameterPreparedStatement(connection, sqlStairwayInstance)) {
-
-      instanceStatement.setString("stairwayName", stairwayName);
-
-      try (ResultSet rs = instanceStatement.getPreparedStatement().executeQuery()) {
-        List<String> stairwayIdList = new ArrayList<>();
-        while (rs.next()) {
-          stairwayIdList.add(rs.getString("stairway_id"));
-        }
-
-        if (stairwayIdList.size() == 0) {
-          return null;
-        }
-        if (stairwayIdList.size() > 1) {
-          throw new DatabaseOperationException("Multiple stairways with the same name!");
-        }
-        return stairwayIdList.get(0);
-      }
-    }
-  }
-
-  /**
-   * This call can be used in conjunction with a list of the currently active stairway instances to
-   * drive recovery of orphaned flights.
+   * Record a new flight
    *
-   * @return list of the stairway instances known to stairway
-   * @throws DatabaseOperationException on SQL exception
+   * @param flightContext description of the flight
+   * @throws DatabaseOperationException database error
+   * @throws InterruptedException thread shutdown
    */
-  List<String> getStairwayInstanceList() throws DatabaseOperationException {
-    final String sql = "SELECT stairway_name FROM " + STAIRWAY_INSTANCE_TABLE;
-    List<String> instanceList = new ArrayList<>();
-
-    try (Connection connection = dataSource.getConnection()) {
-      startReadOnlyTransaction(connection);
-
-      try (NamedParameterPreparedStatement instanceStatement =
-              new NamedParameterPreparedStatement(connection, sql);
-          ResultSet rs = instanceStatement.getPreparedStatement().executeQuery()) {
-        while (rs.next()) {
-          instanceList.add(rs.getString("stairway_name"));
-        }
-      }
-
-      commitTransaction(connection);
-    } catch (SQLException ex) {
-      throw new DatabaseOperationException("Stairway instance enumeration failed", ex);
-    }
-    return instanceList;
-  }
-
-  /** Record a new flight */
   void submit(FlightContext flightContext)
       throws DatabaseOperationException, InterruptedException, DuplicateFlightIdSubmittedException {
+    DbRetry dbRetry = new DbRetry("flight.submit");
```

Comment:
For the pattern `new Foo().method()` (that is, the case where you create an object for the sole purpose of calling a single method on it) I think it's clearer to use a static method on the object. You can still create an object in that static method, but the caller doesn't need to know about this detail.